x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_package_custom_extension/endpoint_package_custom_extension.tsx

```diff
@@ -27,91 +30,155 @@ import {
+          </EuiFlexGroup>
+        );
+      }
+
+      if (!userCanAccessContent) {
+        return <NoPrivileges documentationUrl={docLinks.links.securitySolution.privileges} />;
+      }
+
+      return (
         <div data-test-subj="fleetEndpointPackageCustomContent">
-          <FleetArtifactsCard
-            {...props}
-            artifactApiClientInstance={trustedAppsApiClientInstance}
-            getArtifactsPath={getTrustedAppsListPath}
-            labels={TRUSTED_APPS_LABELS}
-            data-test-subj="trustedApps"
-          />
-          <EuiSpacer />
-          <FleetArtifactsCard
-            {...props}
-            artifactApiClientInstance={eventFiltersApiClientInstance}
-            getArtifactsPath={getEventFiltersListPath}
-            labels={EVENT_FILTERS_LABELS}
-            data-test-subj="eventFilters"
-          />
+          {canReadTrustedApplications && (
+            <>
+              <TrustedAppsArtifactCard {...props} />
+              <EuiSpacer />
+            </>
+          )}
+
+          {canReadEventFilters && (
+            <>
+              <EventFiltersArtifactCard {...props} />
+              <EuiSpacer />
+            </>
+          )}
+
           {canReadHostIsolationExceptions && (
             <>
+              <HostIsolationExceptionsArtifactCard {...props} />
               <EuiSpacer />
-              <FleetArtifactsCard
-                {...props}
-                artifactApiClientInstance={hostIsolationExceptionsApiClientInstance}
-                getArtifactsPath={getHostIsolationExceptionsListPath}
-                labels={HOST_ISOLATION_EXCEPTIONS_LABELS}
-                data-test-subj="hostIsolationExceptions"
-              />
             </>
           )}
-          <EuiSpacer />
-          <FleetArtifactsCard
-            {...props}
-            artifactApiClientInstance={blocklistsApiClientInstance}
-            getArtifactsPath={getBlocklistsListPath}
-            labels={BLOCKLISTS_LABELS}
-            data-test-subj="blocklists"
-          />
+
+          {canReadBlocklist && <BlockListArtifactCard {...props} />}
         </div>
-      ),
-      [
-        blocklistsApiClientInstance,
-        canReadHostIsolationExceptions,
-        eventFiltersApiClientInstance,
-        hostIsolationExceptionsApiClientInstance,
-        trustedAppsApiClientInstance,
-        props,
-      ]
-    );
-
-    return loading ? (
-      <EuiLoadingSpinner data-test-subj="endpointExtensionLoadingSpinner" />
-    ) : canAccessEndpointManagement ? (
-      artifactCards
-    ) : (
-      <NoPermissions />
-    );
+      );
+    }, [
+      canReadBlocklist,
+      canReadEventFilters,
+      canReadTrustedApplications,
+      canReadHostIsolationExceptions,
+      docLinks.links.securitySolution.privileges,
+      loading,
+      props,
+      userCanAccessContent,
+    ]);
+
+    if (loading) {
+      return <EuiLoadingSpinner data-test-subj="endpointExtensionLoadingSpinner" />;
+    }
```

Comment:
Do we really need this loader as well? If we do then can we use the same loader as the one on line 126. Else even though they are the same loaders, they shift slightly from left to center.